components/camel-ribbon/src/test/java/org/apache/camel/component/ribbon/cloud/SpringBeanServiceCallRouteTest.java

```diff
@@ -42,5 +42,13 @@ public void testServiceCallConfiguration() throws Exception {
         RibbonServiceLoadBalancer loadBalancer = (RibbonServiceLoadBalancer)processor.getLoadBalancer();
         Assert.assertTrue(loadBalancer.getServiceDiscovery() instanceof StaticServiceDiscovery);
     }
+    
+    protected String getFirstPort() {
+        return "9092";
```

Comment:
What do you think about using `ServerSocket` to get an available port. Something like:

```
    public static int getAvailablePort(int min, int max) {
        for (int i = min; i <= max; i++) {
            try (ServerSocket socket = new ServerSocket(i)) {
                return socket.getLocalPort();
            } catch (Exception e) {
                System.err.println("Port " + i + " not available, trying next one");
                continue; // try next port
            }
        }
        throw new IllegalStateException("Can't find available network ports");
    }
```

It would create more robust tests, whatever ports are taken on the machine.

Thoughts ?

Reply:
I guess there is also  AvailablePortFinder as part of the camel test suite. 
The problem is that you need to reuse such ports in the xml, @apupier can you see what can be done ?

Reply:
maybe you can use properties ?

Reply:
It makes sense to use properties in this case